python/python/lancedb/query.py

```diff
@@ -1794,27 +2093,64 @@ def nearest_to_text(
         """
         if isinstance(columns, str):
             columns = [columns]
-        return AsyncFTSQuery(
-            self._inner.nearest_to_text({"query": query, "columns": columns})
-        )
+        return AsyncFTSQuery.from_text_query(self, query, columns)
 
 
 class AsyncFTSQuery(AsyncQueryBase):
     """A query for full text search for LanceDB."""
 
-    def __init__(self, inner: LanceFTSQuery):
-        super().__init__(inner)
-        self._inner = inner
+    @dataclass
+    class _FTSArgs:
+        query: Union[str, Awaitable[Optional[str]]]
+        columns: Optional[List[str]] = None
+
+    @classmethod
+    def from_text_query(
+        self,
+        base: AsyncQuery,
+        query: Union[str, Awaitable[Optional[str]]],
+        columns: Optional[List[str]] = None,
+    ):
+        return self(base._table, base._base_args, self._FTSArgs(query, columns))
+
+    def __init__(
+        self,
+        table: "AsyncTable",
+        base_args: AsyncQueryBase._BaseArgs,
+        fts_args: _FTSArgs,
+    ):
+        AsyncQueryBase.__init__(self, table, base_args)
+        self._fts_args = fts_args
+        # TODO: re-ranker
         self._reranker = None
 
-    def get_query(self):
-        self._inner.get_query()
+    def get_query(self) -> str:
+        if isinstance(self._fts_args.query, Awaitable):
+            raise ValueError("The query is an Awaitable, please call _resolve() first.")
+        return self._fts_args.query
+
+    async def _resolve(self) -> LanceFTSQuery:
+        if isinstance(self._fts_args.query, Awaitable):
+            query = await self._fts_args.query
+            if query is None:
+                # Drop down to plain query
+                return AsyncQuery(self._table, self._base_args)
+            else:
+                self._fts_args.query = query
+        return self
+
+    def _to_inner(self) -> LanceQuery:
+        inner = super()._to_inner()
+        inner = inner.nearest_to_text(
+            dict(query=self._fts_args.query, columns=self._fts_args.columns or [])
+        )
+        return inner
 
     def rerank(
         self,
-        reranker: Reranker = RRFReranker(),
+        reranker: Reranker,
```

Comment:
I'm not sure how the tests were passing, but `RRFReranker()` doesn't support pure FTS queries or pure vector queries. Therefore, I've removed it as the default for these.